### PR TITLE
Add index URLs when provided via `uv add --index` or `--default-index`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

The behavior is as follows:

- If you provide `--index` or `--default-index` on the command-line, we add those indexes to the `pyproject.toml` (with names, if provided, as in `--index pytorch=https://download.pytorch.org/whl/cu121`.
- If you provide `--index-url` or `--default-index`, we warn, but don't add the indexes to the file. (This seems wrong -- why not add them?)
- If you provide an index with a name or URL that already exists, we remove that entry, and add the new index to the top of the list (since it now has highest priority).
- If you provide a `--default-index`, and an index already has `default = true`, we remove that entry, since it won't be used anymore.

We do _not_ pin packages to specific indexes yet.

### CODE REVIEW

**Review by konstin — APPROVED** [review]

File: crates/uv/tests/edit.rs [hunk 1]
```diff
@@ -3707,7 +3707,7 @@ fn add_lower_bound_local() -> Result<()> {
     "#})?;
 
     // Adding `torch` should include a lower-bound, but no local segment.
-    uv_snapshot!(context.filters(), context.add().arg("local-simple-a").arg("--extra-index-url").arg(packse_index_url()).env_remove("UV_EXCLUDE_NEWER"), @r###"
+    uv_snapshot!(context.filters(), context.add().arg("local-simple-a").arg("--index").arg(packse_index_url()).env_remove("UV_EXCLUDE_NEWER"), @r###"
```
[the remark this answers is not in the document]
Comment (charliermarsh) [reply crates/uv/tests/edit.rs:3710]: As in, we should require it?

[the remark this answers is not in the document]
Comment (konstin) [reply crates/uv/tests/edit.rs:3710]: Yeah

[the remark this answers is not in the document]
Comment (zanieb) [reply crates/uv/tests/edit.rs:3710]: What's the benefit of requiring it? It seems restrictive.

[the remark this answers is not in the document]
Comment (konstin) [reply crates/uv/tests/edit.rs:3710]: Consistency and the option to refer to an index by name everywhere

[the remark this answers is not in the document]
Comment (zanieb) [reply crates/uv/tests/edit.rs:3710]: Oh you think we should require a name even when they're defined in the file manually? I'm not sure it's worth it.

**Pull request merged**